inference-engine/src/gna_plugin/transformations/convert_matmul_to_pointwise_convolution.cpp

```diff
@@ -0,0 +1,135 @@
+    if (input1_shape.front() <= GNALimitations::affineMaxBatchSize ||
+        out_channels % GNALimitations::convFiltersNumDivider != 0 ||
+        out_channels > GNALimitations::convMaxFiltersNum ||
+        in_channels > GNALimitations::convFilterMaxSize) {
+        return std::make_tuple(false, 0, 0, 0);
+    }
+
+    return std::make_tuple(true, width, in_channels, out_channels);
+}
+
+ConvertMatmulToPointWiseConvolution::ConvertMatmulToPointWiseConvolution() {
+    auto const_input = ngraph::pattern::wrap_type<ngraph::opset7::Constant>();
+    auto const_fq = ngraph::pattern::wrap_type<ngraph::opset7::FakeQuantize>({const_input,
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>()});
+    auto second_input = std::make_shared<ngraph::pattern::op::Or>(ngraph::OutputVector{const_input, const_fq});
+    auto matmul = ngraph::pattern::wrap_type<ngraph::opset7::MatMul>({ngraph::pattern::any_input(), second_input});
+    auto add = ngraph::pattern::wrap_type<ngraph::opset7::Add>({matmul,
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>()});
+    auto matmul_out = std::make_shared<ngraph::pattern::op::Or>(ngraph::OutputVector{matmul, add});
+    auto out_fq = ngraph::pattern::wrap_type<ngraph::opset7::FakeQuantize>({matmul_out,
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>()});
+    auto root = std::make_shared<ngraph::pattern::op::Or>(ngraph::OutputVector{matmul_out, out_fq});
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto& pattern_map = m.get_pattern_value_map();
+        auto matmul_node = pattern_map.at(matmul).get_node_shared_ptr();
+        bool supported;
+        uint32_t width, in_channels, out_channels;
+        std::tie(supported, width, in_channels, out_channels) = VerifyAndGetConvParams(matmul_node);
+        if (!supported) return false;
+
+        auto input_node = matmul_node->input_value(0).get_node_shared_ptr();
+        auto weights_node = matmul_node->input_value(1).get_node_shared_ptr();
+        std::shared_ptr<ngraph::Node> matmul_output =
+            matmul_node->output(0).get_target_inputs().begin()->get_node()->shared_from_this();
+        std::shared_ptr<ngraph::Node> bias_node = nullptr;
+        if (std::dynamic_pointer_cast<ngraph::opset7::Add>(matmul_output)) {
+            auto add_second_input = matmul_output->input_value(1).get_node_shared_ptr();
+            if (std::dynamic_pointer_cast<ngraph::opset7::Constant>(matmul_output)) {
+                bias_node = matmul_output->input_value(1).get_node_shared_ptr();
```

Comment:
Use Output<Node> instead of Node pointer. If this node is multioutput you will get an exception.
```suggestion
                bias_node = matmul_output->input_value(1);
```

Reply:
get bias using pattern_map